container/rules/docker_toolchains.bzl

```diff
@@ -308,54 +348,79 @@ def _toolchain_container_impl(ctx):
     keys = depset(keys).to_list()
     installables_tars = depset(installables_tars).to_list()
 
-    return _language_tool_layer_impl(
-        ctx,
-        symlinks = symlinks,
-        env = env,
-        tars = tars,
-        files = files,
-        packages = packages,
-        additional_repos = additional_repos,
-        keys = keys,
-        installables_tars = installables_tars,
-        installation_cleanup_commands = installation_cleanup_commands,
-    )
+    # If ctx.file.installables_tar is specified, ignore 'packages' and other tars in installables_tars.
+    if ctx.file.installables_tar:
```

Comment:
It does not make much sense to have this logic inside _debian_pkg_tar_impl. This belongs in the toolchain container rule (or the language_tool_layer rule), which should call the _debian_pkg_tar_impl only if it needs to (i.e., if ctx.file.installables_tar is not specified)

Reply:
Yea, you are right. I will move this logic into the implementation of ```toolchain_container``` rule.